strawberryfields/backends/states.py

```diff
@@ -914,6 +914,26 @@ def number_expectation(self, modes):
             ps = np.tensordot(np.diag(values), ps, axes=((0, 1), (0, 1)))
         return float(ps)
 
+    def parity_expectation(self, mode):
+
+        if (type(mode) != int):
+            raise ValueError("Specified mode index must be an integer.")
+
+        cutoff = self._cutoff  # Fock space cutoff.
+        num_modes = self._modes # number of modes in the state.
+        values = np.arange(cutoff)
+
+        print(self.is_pure)
+
+        if self.is_pure:
```

Comment:
@Lucaman99 this would probably make a better addition if you also allow for mixed state for which you can also get the Fock probabilities.